### PR TITLE
[callbacks] Store file/func info in callbacks for debug build

### PR DESCRIPTION
**Pull request by jprestwo**
Signed-off-by: James Prestwood <james.prestwood@intel.com>

### CODE REVIEW

File: src/zjs_callbacks.c [hunk 1]
```diff
@@ -383,11 +423,18 @@ void zjs_remove_all_callbacks()
 }
 
 // INTERRUPT SAFE FUNCTION: No JerryScript VM, allocs, or release prints!
-void zjs_signal_callback(zjs_callback_id id, const void *args, uint32_t size)
+void signal_callback_priv(zjs_callback_id id, const void *args, uint32_t size
+#ifdef DEBUG_BUILD
+        , const char *file, const char *func)
```
Comment (jimmy-huang) [inline src/zjs_callbacks.c]: align indentation

File: src/zjs_callbacks.h [hunk 2]
```diff
@@ -149,6 +102,14 @@ void zjs_remove_callback(zjs_callback_id id);
  */
 void zjs_remove_all_callbacks();
 
+void signal_callback_priv(zjs_callback_id id, const void *args, uint32_t size
+#ifdef DEBUG_BUILD
+        , const char *file, const char *func);
```
Comment (jimmy-huang) [inline src/zjs_callbacks.h]: align indentation

File: src/zjs_callbacks.h [hunk 3]
```diff
@@ -170,7 +131,90 @@ void zjs_remove_all_callbacks();
...
+    add_callback_priv(func, this, handle, post, 1);
+#else
+#define zjs_add_callback(func, this, handle, post) \
+    add_callback_priv(func, this, handle, post, 0, __FILE__, __func__)
+#define zjs_add_callback_once(func, this, handle, post) \
+    add_callback_priv(func, this, handle, post, 1, __FILE__, __func__);
+#endif
+
+zjs_callback_id add_callback_list_priv(jerry_value_t js_func,
+                                      jerry_value_t this,
```
Comment (jimmy-huang) [inline src/zjs_callbacks.h]: align indentation

File: src/zjs_callbacks.c [hunk 4]
```diff
@@ -90,6 +94,21 @@ static int zjs_ringbuf_error_count = 0;
...
+{
+    uint32_t file_len = strlen(file);
+    uint32_t func_len = strlen(func);
+    *str = zjs_malloc(file_len + func_len + 4);
+    strncpy(*str, file, file_len);
+    (*str)[file_len] = ':';
+    strncpy(*str + file_len + 1, func, func_len);
+    (*str)[file_len + func_len + 1] = '(';
+    (*str)[file_len + func_len + 2] = ')';
+    (*str)[file_len + func_len + 3] = '\0';
```
Comment (grgustaf) [inline src/zjs_callbacks.c]: Check malloc rval. Probably still set the *str to NULL in that case, just make sure you handle that elsewhere. I think printf("%s", NULL) works okay, just ugly but okay for debug in this rare case.

For readability, I suggest declare a char * and then assign *str to it at the end. Although after the following it won't matter as much.

And then replace most of this with:
snprintf("%s:%s()", file, func);
... FTW!

**Comment by grgustaf** [comment]
+1

**Pull request merged**